rfc/rfc-89/rfc-89.md

```diff
@@ -0,0 +1,345 @@
+             "fields": [
+               {
+                 "name": "bucketNumber",
+                 "type": ["null", "int"],
+                 "default": null
+               },
+               {
+                 "name": "instant",
+                 "type":["null","string"],
+                 "default": null
+               }
+             ]
+           }}],
+       "default" : null
+     }
+   ]
+}
+```
+We will write <instant>.hashing_meta during commit action (at pre-commit stage):
+1. Get latest complete T1.hashing_meta
+2. Merging T1.hashing_meta with new written partitions as T2.hashing_meta also do partition conflict detection
+   1. For example, current write write partition1 with 20 buckets but there is already defined partition1 with 10 bucket numbers, which means there is 
+another writer that uses the latest expression to pre-determine the bucket number of the current partition. In order to ensure data consistency, it is necessary to abort the current writer.
+3. If conflict detection is passed AND **new partitions are added** then write T2.hashing_meta into .hoodie folder
+   1. Here, a new hashing_meta will only be written when a new partition is added to avoid unnecessary I/O overhead.
+4. Do commit action
+
+Using Archive Service to clean up expired hashing meta, keeping up to 10 historical hashing meta
+
+### SimpleBucketIdentifier
+```java
+public class SimpleBucketIdentifier extends BucketIdentifier {
+  // use expressionEvaluator to compute bucket number for given partition path
+  private BucketCalculator calculator;
+  // get previous partition and bucket number from meta under .hoodie folder
+  private Map<String, Integer> partition2BucketNumber;
+
+  @Override
+  public int getBucketId(String recordKey, String indexKeyFields, String partitionPath) {
+    int numBuckets = getBucketNumber(partitionPath);
+    return getBucketId(getHashKeys(recordKey, indexKeyFields), numBuckets);
+  }
+  ...
+  
+  public int getBucketNumber(String partitionPath) {
+    return calculator.calculateBucketNumber(partitionPath);
+  }
+}
+```
+**WorkFlow for SimpleBucketIdentifier**
+1. Get and Cache partition2BucketNumber 
+2. For the partition of the given data, try to get its corresponding bucket number from the cache. If the cache missed, 
+   1. Calculate the expression through the Bucket Calculator to obtain the corresponding Bucket Number.
+   2. Put Calculated bucket number into cache
+
+Using SimpleBucketIdentifier to get `BucketNumber` or `BucketID` based on given partitionPath and corresponding expression
+
+For new Partitioner:
+`SparkBucketIndexPartitioner`-Spark and `BucketIndexPartitioner`-Flink use this new SimpleBucketIdentifier to do data repartition
+
+For Writer:
+Writer like `BucketStreamWriteFunction` in Flink also use this new `SimpleBucketIdentifier` to compute bucketID
+
+NOTE：We need to get hashing_meta instant and hashing_config `instant` at the beginning(JM or Driver) and pass it to each operator, 
```

Comment:
Still need a listing action at Master side(JM or Driver) to decide which meta and config need to load.